File: habit-tracker/src/main/java/com/habit/habit_tracker/security/AuthUtil.java
package com.habit.habit_tracker.security;

import org.springframework.http.HttpStatus;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.stereotype.Component;

import com.habit.habit_tracker.domain.User;
import com.habit.habit_tracker.exception.ApiRequestException;

@Component
public class AuthUtil {

    public User getAuthenticatedUser() {
        if (SecurityContextHolder.getContext().getAuthentication() == null ||
                !(SecurityContextHolder.getContext().getAuthentication().getPrincipal() instanceof UserPrincipal)) {
            throw new ApiRequestException("User not authenticated", HttpStatus.UNAUTHORIZED);
        }

        return ((UserPrincipal) SecurityContextHolder.getContext().getAuthentication().getPrincipal()).getUser();
    }
}

File: habit-tracker/src/main/kotlin/com/habit/habit_tracker/security/AuthUtil.kt
package com.habit.habit_tracker.security

import com.habit.habit_tracker.domain.User
import com.habit.habit_tracker.exception.ApiRequestException
import org.springframework.http.HttpStatus
import org.springframework.security.core.context.SecurityContextHolder
import org.springframework.stereotype.Component

@Component
class AuthUtil {

    fun getAuthenticatedUser(): User {
        val authentication = SecurityContextHolder.getContext().authentication
            ?: throw ApiRequestException("User not authenticated", HttpStatus.UNAUTHORIZED)

        val principal = authentication.principal
        if (principal !is UserPrincipal) {
            throw ApiRequestException("User not authenticated", HttpStatus.UNAUTHORIZED)
        }

        return principal.getUser()
    }
}
